src/jbi/router.py

```diff
@@ -1,34 +1,78 @@
 """
 Router dedicated to Jira Bugzilla Integration APIs
 """
+import importlib
 import logging
-from typing import Optional
+from types import ModuleType
+from typing import Dict, List, Optional
 
-from fastapi import APIRouter, Depends, Request
-from fastapi.responses import HTMLResponse
+from fastapi import APIRouter, Body, Depends, Request
+from fastapi.responses import HTMLResponse, JSONResponse
 from fastapi.templating import Jinja2Templates
 
 from src.app import environment
 from src.jbi import configuration
+from src.jbi.bugzilla import BugzillaBug, BugzillaWebhookRequest
+from src.jbi.errors import IgnoreInvalidRequestError
+from src.jbi.services import getbug_as_bugzilla_object
 
 templates = Jinja2Templates(directory="src/templates")
 
 api_router = APIRouter(tags=["JBI"])
 
-jbi_logger = logging.getLogger("src.jbi")
+jbi_logger = logging.getLogger("src.jbi.router")
 
 
-def execute_request(request, settings):
+def extract_current_action(  # pylint: disable=inconsistent-return-statements
+    bug_obj: BugzillaBug, action_map
+):
+    """Find first matching action from bug's whiteboard list"""
+    potential_configuration_tags: List[
+        str
+    ] = bug_obj.get_potential_whiteboard_config_list()
+
+    for tag in potential_configuration_tags:
+        if action := action_map.get(tag.lower()):
+            return action
+
+
+def execute_action(request: BugzillaWebhookRequest, action_map, settings):
     """Execute action"""
+    try:
+        if not request.bug:
+            raise IgnoreInvalidRequestError("no bug data received")
+
+        is_private_bug = request.bug.is_private
+        if is_private_bug:
+            raise IgnoreInvalidRequestError("private bugs are not valid")
+
+        request.bug = getbug_as_bugzilla_object(request=request)
+        current_action = extract_current_action(request.bug, action_map)  # type: ignore
+        if not current_action:
+            raise IgnoreInvalidRequestError(
+                "whiteboard tag not found in configured actions"
+            )
+
+        action_module: ModuleType = importlib.import_module(current_action["action"])
+        callable_action = action_module.init(  # type: ignore
+            **current_action["parameters"]
+        )
+        content = callable_action(payload=request)
+        jbi_logger.info("request: %s, content: %s", request.json(), content)
+        return JSONResponse(content=content, status_code=200)
+    except IgnoreInvalidRequestError as exception:
+        jbi_logger.info("ignore-invalid-request: %s", exception, exc_info=True)
```

Comment:
Are you sure this is going to be too much logging (and same level as executed actions)? 

The `exc_info` gives the stracktrace IIRC but we don't really care here IMO. 

I would suggest to either:
- create a dedicated logger `invalid_logger = logging.getLogging("ignored-requests")` so that we can filter it easily in prod
- switch to debug level here

Reply:
Agreed, I've made the changes in a new PR as a fast follow; I'm going to resolve this and merge this in as-is for now.
https://github.com/mozilla/jira-bugzilla-integration/pull/34